extensions/proposals/WEBGL_element_texture/extension.xml

```diff
@@ -0,0 +1,298 @@
+    <dt><code class="attribute-name">onframeavailable</code> of type
+        <code>EventHandler</code></dt>
+
+    <dd>The <code>onframeavailable</code> handler is executed when a new frame
+        could be latched from image stream.</dd>
+  </dl>
+
+  <!-- new tokens -->
+
+  <newtok>
+    <p>On <code>WEBGL_element_texture</code>:</p>
+
+    <p>The meaning and use of these tokens is similar as described in <a
+    href="https://www.khronos.org/registry/OpenGL/extensions/OES/OES_EGL_image_external.txt">OES_EGL_image_external</a>.</p>
+
+    <function name="bindTexture" type="void">
+      <param name="target" type="GLenum"/>
+      <param name="texture" type="WebGLTexture?"/>
+      <code>TEXTURE_ELEMENT_WEBGL</code> is accepted as a target by the
+      <code>target</code> parameter of <code>bindTexture()</code>
+    </function>
+
+    <function name="getActiveUniform" type="WebGLActiveInfo?">
+      <param name="program" type="WebGLProgram?"/>
+      <param name="index" type="GLuint"/>
+      <code>SAMPLER_ELEMENT_WEBGL</code> can be returned in the
+      <code>type</code> field of the <code>WebGLActiveInfo</code> returned by
+      <code>getActiveUniform()</code>
+    </function>
+
+    <function name="getParameter" type="any">
+      <param name="pname" type="GLenum"/>
+      <code>TEXTURE_BINDING_ELEMENT_WEBGL</code> is accepted by
+      the <code>pname</code> parameter of <code>getParameter()</code>
+    </function>
+
+    <function name="getTexParameter*" type="any">
+      <param name="target" type="GLenum"/>
+      <param name="pname" type="GLenum"/>
+      <code>REQUIRED_TEXTURE_ELEMENT_IMAGE_UNITS_WEBGL</code> is accepted
+      as the <code>pname</code> parameter of <code>getTexParameter*()</code>
+    </function>
+  </newtok>
+
+  <samplecode xml:space="preserve">
+
+    <p> This a fragment shader that samples a element texture.</p>
+    <pre>
+    #extension GL_WEBGL_element_texture : require
+    precision mediump float;
+    varying vec2 v_texCoord;
+
+    uniform samplerElementWEBGL uSampler;
+
+    void main(void) {
+      gl_FragColor = texture2D(uSampler, v_texCoord);
+    }
+    </pre>
+
+    <p>This shows application that renders HTML*Element using proposed extension.</p>
+    <pre>
+    var customElement = document.getElementById("...");
+    var customTexture = gl.createTexture();
+    var elementTexture;
+
+    function initialize() {
+      var ext = gl.getExtension('WEBGL_element_texture');
+      if (ext === null)
+        return;
+
+      elementTexture = ext.create*Texture(customElement, customTexture);
+      elementTexture.onframeavailable = function() { ... };
+    }
+
+    function update() {
+      gl.bindTexture(ext.TEXTURE_ELEMENT_WEBGL, customTexture);
+      elementTexture.updateTexImage();
+      gl.bindTexture(ext.TEXTURE_ELEMENT_WEBGL, null);
+    }
+
+    function render() {
+        .....
+
+        gl.activeTexture(gl.TEXTURE0);
+        gl.bindTexture(ext.TEXTURE_ELEMENT_WEBGL, customTexture);
```

Comment:
It seems unnecessary to keep binding and unbinding this customTexture. Unless the underlying implementation requires it, we shouldn't do it.

Reply:
I added this code to mention how we will use this texture with new target in the rendering stage.
No point to keep binding and unbinding this texture all time. I will remove these lines.